src/cargo/util/config/mod.rs

```diff
@@ -873,9 +873,17 @@ impl Config {
             return Ok(Some(path));
         }
 
-        let var = format!("build.{}", tool);
-        if let Some(tool_path) = self.get_path(&var)? {
-            return Ok(Some(tool_path.val));
+        // For backwards compatibility we allow both snake_case config paths as well as the
+        // idiomatic kebab-case paths.
+        let config_paths = [
+            format!("build.{}", tool),
+            format!("build.{}", tool.replace('_',  "-"))
+        ];
```

Comment:
Not sure how important it is to keep backwards compatibility here, since this is not a documented feature.  If it is not important I can get rid of this code.